client/app/src/main/java/com/example/criengine/Activities/UserProfileActivity.java

```diff
@@ -18,10 +49,41 @@
      */
     @Override
     protected void onCreate(Bundle savedInstanceState) {
-        setContentView(R.layout.activity_user_profile);
         super.onCreate(savedInstanceState);
 
-        // new UI components
+        if (getIntent().getExtras() != null) {
+            userId = getIntent().getStringExtra("userId");
+        } else {
+            Intent intent = new Intent(this, SomethingWentWrong.class);
+            startActivity(intent);
+            return;
+        }
+
+        userBooksTextView = findViewById(R.id.user_books_text);
+        userBooksListView = findViewById(R.id.user_books_listview);
```

Comment:
can we also get the EditTextViews of: "user_profile_text", "bio_text", "phone_text", "addr_text"
and make it so they are not editable... ie https://github.com/CMPUT301F20T05/CriEngine/blob/07153950cd56a1b526ad05d34209142deb45fada/client/app/src/main/java/com/example/criengine/Fragments/MyProfileFragment.java#L35-L46